File: adding_one/JS/addingOne.js
console.log("Adding One in JS!");

// "Increments" an array
const addingOne = (array, indx)=> {
	// Start from end of array
	if(indx==null) indx = array.length - 1;

	// Base for reccursion
	// Insert 1 at the beginning of the array 1. (Adding digit)
	if(indx < 0) {
		array.unshift(1);
		return array;
	}

	// Carry the one using reccursion.
	if(array[indx] >= 9) {
		array[indx] = 0;
		return addingOne(array, --indx);
	}
	else
		array[indx]++;
	return array;
}

console.log(addingOne([1, 2, 3, 4])); // 1,235

console.log(addingOne([1, 2, 9, 9])); // 1,300

console.log(addingOne([9, 9, 9, 9])); // 10,000

File: adding_one/Python/adding_one.py
print("Adding One With Python!")

def adding_one(array, indx=None):

	# Start at end of array
	if(indx == None):
		indx = len(array) - 1

	# Exit code for Reccursion (Insert 1 to begging of number)
	if(indx < 0):
		array.insert(0, 1)
		return array

	# Carry the 1 if digit equals 9
	if(array[indx] >= 9):
		array[indx] = 0
		indx-=1
		return adding_one(array, indx)

	# Else just add one
	else:
		array[indx]+=1
	return array


print(adding_one([1,2,3])) # 124

print(adding_one([9,9,9])) # 1,000

print(adding_one([6,5,9,9,0])) # 65,991